lib/classes/Variables.test.js

```diff
@@ -875,67 +922,72 @@ describe('Variables', () => {
     });
   });
 
-  describe('#warnIfNotFound()', () => {
-    let logWarningSpy;
-    let consoleLogStub;
-    let varProxy;
-
-    beforeEach(() => {
-      logWarningSpy = sinon.spy(slsError, 'logWarning');
-      consoleLogStub = sinon.stub(console, 'log').returns();
-      const ProxyQuiredVariables = proxyquire('./Variables.js', {
-        './Error': logWarningSpy,
-      });
-      varProxy = new ProxyQuiredVariables(new Serverless());
-    });
+  describe('#throwIfNotFound()', () => {
+    const serverless = new Serverless();
 
-    afterEach(() => {
-      logWarningSpy.restore();
-      consoleLogStub.restore();
+    it('should do nothing if variable value is a string.', () => {
+      expect(() => serverless.variables.throwIfNotFound('self:service', 'a-valid-value'))
+        .to.not.throw(Error);
     });
 
-    it('should do nothing if variable has valid value.', () => {
-      varProxy.warnIfNotFound('self:service', 'a-valid-value');
-      expect(logWarningSpy.calledOnce).to.be.equal(false);
+    it('should do nothing if variable value is a number.', () => {
+      expect(() => serverless.variables.throwIfNotFound('self:service', 0))
+        .to.not.throw(Error);
     });
 
-    it('should log if variable has null value.', () => {
-      varProxy.warnIfNotFound('self:service', null);
-      expect(logWarningSpy.calledOnce).to.be.equal(true);
+    it('should do nothing if variable value is null.', () => {
+      expect(() => serverless.variables.throwIfNotFound('self:service', null))
+        .to.not.throw(Error);
     });
 
-    it('should log if variable has undefined value.', () => {
-      varProxy.warnIfNotFound('self:service', undefined);
-      expect(logWarningSpy.calledOnce).to.be.equal(true);
+    it('should throw if variable value is undefined.', () => {
+      expect(() => serverless.variables.throwIfNotFound('self:service', undefined))
+        .to.throw(Error);
     });
 
-    it('should log if variable has empty object value.', () => {
-      varProxy.warnIfNotFound('self:service', {});
-      expect(logWarningSpy.calledOnce).to.be.equal(true);
+    it('should throw if variable value is an empty object.', () => {
+      expect(() => serverless.variables.throwIfNotFound('self:service', {}))
+        .to.throw(Error);
     });
 
     it('should detect the "environment variable" variable type', () => {
-      varProxy.warnIfNotFound('env:service', null);
-      expect(logWarningSpy.calledOnce).to.be.equal(true);
-      expect(logWarningSpy.args[0][0]).to.contain('environment variable');
+      expect(() => serverless.variables.throwIfNotFound('env:service', undefined))
+        .to.throw(Error);
```

Comment:
The block can be just be
```
expect(() => serverless.variables.throwIfNotFound('env:service', undefined))
       .to.throw(/environment variable/);
```

Same in the tests below. This eliminates the need of any try/catch blocks.